rules/DowngradePhp70/Rector/Instanceof_/DowngradeInstanceofThrowableRector.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp70\Rector\Instanceof_;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\BinaryOp\BooleanOr;
+use PhpParser\Node\Expr\Instanceof_;
+use PhpParser\Node\Name;
+use PhpParser\Node\Name\FullyQualified;
+use Rector\Core\Rector\AbstractRector;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * Throwable was introduced in PHP 7.0 so to support older versions we need to also check for Exception.
+ * @see https://www.php.net/manual/en/class.throwable.php
+ * @see \Rector\Tests\DowngradePhp70\Rector\Instanceof_\DowngradeInstanceofThrowableRector\DowngradeInstanceofThrowableRectorTest
+ */
+final class DowngradeInstanceofThrowableRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Add `instanceof Exception` check as a fallback to `instanceof Throwable` to support exception hierarchies in PHP 5', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+return $e instanceof \Throwable;
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+return $e instanceof \Throwable || $e instanceof \Exception;
```

Comment:
I also wonder about producing code like `getExceptionWithSideEffects() instanceof \Throwable || getExceptionWithSideEffects() instanceof \Exception`. Perhaps I should move it to a temporary variable (but how do I come up with a unique name?)

Reply:
You can use VariableNaming service to create unique variable, eg:

https://github.com/rectorphp/rector-src/blob/fc10fce13dcf9767f54e4202b509020fed338645/rules/DowngradePhp55/Rector/Foreach_/DowngradeForeachListRector.php#L84